src/3d/terrain/qgsdemterraintileloader_p.h

```diff
@@ -0,0 +1,114 @@
+#ifndef QGSDEMTERRAINTILELOADER_P_H
+#define QGSDEMTERRAINTILELOADER_P_H
+
+///@cond PRIVATE
+
+//
+//  W A R N I N G
+//  -------------
+//
+// This file is not part of the QGIS API.  It exists purely as an
+// implementation detail.  This header file may change from version to
+// version without notice, or even be removed.
+//
+
+#include <QtConcurrent/QtConcurrentRun>
+#include <QFutureWatcher>
+#include <QElapsedTimer>
+
+#include "qgsrectangle.h"
+#include "qgsterraintileloader_p.h"
+#include "qgstilingscheme.h"
+
+class QgsRasterDataProvider;
+class QgsRasterLayer;
+
+/** \ingroup 3d
+ * Chunk loader for DEM terrain tiles.
+ * \since QGIS 3.0
+ */
+class QgsDemTerrainTileLoader : public QgsTerrainTileLoader
+{
+    Q_OBJECT
+  public:
+    //! Constructs loader for the given chunk node
+    QgsDemTerrainTileLoader( QgsTerrainEntity *terrain, QgsChunkNode *node );
+    ~QgsDemTerrainTileLoader();
+
+    virtual Qt3DCore::QEntity *createEntity( Qt3DCore::QEntity *parent );
```

Comment:
SIP_FACTORY ?

Reply:
This is meant to stay private class (_p naming), so I would not put SIP annotations there...